### ISSUE TITLE
Cinnamon settings daemon blocks systemd lid-switch action

### ISSUE THREAD

**Issue by marmistrz** [linuxmint/cinnamon-settings-daemon#141]
I'm using cinnamon-settings-daemon 3.2.0-1 under Arch Linux. I just noticed that my laptop doesn't go to sleep. The cause of the problem is that cinnamon-settings-daemon inhibits handle-lid-switch, as `systemd-inhibit` reports:
```
systemd-inhibit
     Who: NetworkManager (UID 0/root, PID 306/NetworkManager)
    What: sleep
     Why: NetworkManager needs to turn off networks
    Mode: delay

     Who: marcin (UID 1000/marcin, PID 937/cinnamon-settin)
    What: handle-lid-switch
     Why: Multiple displays attached
    Mode: block

     Who: marcin (UID 1000/marcin, PID 937/cinnamon-settin)
    What: sleep
     Why: Cinnamon needs to lock the screen
    Mode: delay

     Who: marcin (UID 1000/marcin, PID 937/cinnamon-settin)
    What: handle-power-key:handle-suspend-key:handle-hibernate-key
     Why: Cinnamon handling keypresses
    Mode: block

4 inhibitors listed.
```

I want systemd to handle the lid switch action because, unlike Cinnamon, it allows me to temporarily inhibit the lid switch (via `systemd-inhibit`). AFAIK Cinnamon doesn't provide such functionality.

At least, disabling the systemd things should be voluntary.

**Comment by marmistrz** [comment]
It's very important to me, so I'll make a fix myself and send a pull request, just tell me, where the relvant sources reside :)

**Comment by leigh123linux** [comment]
Is this what your to lazy to look for? 😄 

https://github.com/linuxmint/cinnamon-settings-daemon/blob/master/plugins/power/csd-power-manager.c#L3903

**Comment by marmistrz** [comment]
Well, I looked for it. I thought cinnamon-settings was the same as cinnamon-settings-daemon. Now I see it's not ;)

**Comment by marmistrz** [comment]
Has anyone worked on it? I started playing around but had no time and probably won't have any until July

**Comment by dr0bz** [comment]
I hope this is the same issue, using Cinnamon 3.4.1.

On lid close it does not suspend and i get following:

`Mai 29 12:17:50 xmframe csd-power[2658]: couldn't sleep using UPower: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: Keine derartige Methode »Suspend« Mai 29 12:17:51 xmframe systemd-logind[1104]: Lid opened. Mai 29 12:17:51 xmframe repowerd[1102]: UPowerPowerSourceAndLid: change_upower(), lid_is_closed=false`

systemd-inhibit

    Who: gdm (UID 135/gdm, PID 2103/gsd-media-keys)
    What: handle-power-key:handle-suspend-key:handle-hibernate-key
     Why: GNOME handling keypresses
    Mode: block

     Who: hermann (UID 1000/hermann, PID 2645/csd-media-keys)
    What: handle-power-key:handle-suspend-key:handle-hibernate-key
     Why: Cinnamon handling keypresses
    Mode: block

     Who: hermann (UID 1000/hermann, PID 2716/gsd-media-keys)
    What: handle-power-key:handle-suspend-key:handle-hibernate-key
     Why: GNOME handling keypresses
    Mode: block

     Who: hermann (UID 1000/hermann, PID 2658/csd-power)
    What: sleep
     Why: Cinnamon needs to lock the screen
    Mode: delay

     Who: Telepathy (UID 1000/hermann, PID 2839/mission-control)
    What: shutdown:sleep
     Why: Disconnecting IM accounts before suspend/shutdown...
    Mode: delay

     Who: hermann (UID 1000/hermann, PID 2658/csd-power)
    What: handle-lid-switch
     Why: Multiple displays attached
    Mode: block

     Who: ModemManager (UID 0/root, PID 1095/ModemManager)
    What: sleep
     Why: ModemManager needs to reset devices
    Mode: delay

     Who: NetworkManager (UID 0/root, PID 1093/NetworkManager)
    What: sleep
     Why: NetworkManager needs to turn off networks
    Mode: delay

8 inhibitors listed.

**Comment by leigh123linux** [comment]
@dr0bz Can you post

```
gsettings list-recursively org.cinnamon.desktop.session
```

**Comment by dr0bz** [comment]
@leigh123linux 
`org.cinnamon.desktop.session session-manager-uses-logind false
org.cinnamon.desktop.session session-name 'cinnamon'
org.cinnamon.desktop.session settings-daemon-uses-logind false
org.cinnamon.desktop.session idle-delay uint32 0
org.cinnamon.desktop.session screensaver-uses-logind false`

**Comment by leigh123linux** [comment]
Try using logind as your upower version has obsoleted suspend/hibernate

```
gsettings set org.cinnamon.desktop.session session-manager-uses-logind true
gsettings set org.cinnamon.desktop.session settings-daemon-uses-logind true
gsettings set org.cinnamon.desktop.session screensaver-uses-logind true
```

**Comment by dr0bz** [comment]
@leigh123linux hey thanks! It works.

**Comment by LinuxOnTheDesktop** [comment]
Having read this thread I am unsure whether (1) there was a real problem here and if so (2) whether it was fixed.

I mention those things for the following reason. I have a ThinkPad that runs Mint Cinnamon and which, when on battery power, does not suspend when the lid is closed. I have tried to fix the problem by having a script run the following before every suspend. I am unsure whether the fix always works.

```
for device in XHC PWRB; do
	grep -q -m 1 "${device}.*enabled" /proc/acpi/wakeup &&
		printf "%s" ${device} >/proc/acpi/wakeup
done
```

But perhaps this is a hardware or firmware problem; I got the above from [here](https://unix.stackexchange.com/questions/448974/laptop-wake-up-when-lid-is-closed#449861).
